frontend/src/metabase/dashboard/actions/data-fetching.js

```diff
@@ -389,6 +390,8 @@ export const fetchDashboardCardData = createThunkAction(
 
     dispatch(setDocumentTitle(t`0/${promises.length} loaded`));
 
+    // XXX: There is a race condition here, when refreshing a dashboard before
```

Comment:
Added a comment, so we are aware there is a race condition here. This happens when we refresh a dashboard before the previous API call finish. e.g. when we set a parameter value before the dashboard is completely loaded. It could show the stale result while the new one is loading.

Ideally, the old promises should be rejected. When the new ones are created.